src/checkout/hooks/useCheckoutStepState.ts

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from "react";
+
+import { VariantsProducts_productVariants } from "@sdk/queries/types/VariantsProducts";
+
+import { CardData } from "../types/CardData";
+import { Checkout } from "../types/Checkout";
+
+export enum CheckoutStep {
+  ShippingAddress = 1,
+  ShippingOption,
+  BillingAddress,
+  Payment,
+  Review,
+}
+
+export const useCheckoutStepState = (
+  checkout: Checkout,
+  variantsProducts: VariantsProducts_productVariants,
+  cardData: CardData,
+  dummyStatus: string
+): CheckoutStep => {
+  const isShippingRequiredForProducts = () => {
+    const isShippingRequired =
```

Comment:
I think this can be simplified to just
```
return variantsProducts.edges && variantsProducts.edges.some(
        ({ node }) => node.product.productType.isShippingRequired
);
```

If there are no `edges` we will get false, and `some` is returning Boolean so we dont need temp variable for keeping the result

Reply:
Sure, fixed